pandas/core/indexes/datetimes.py

```diff
@@ -915,46 +915,6 @@ def to_series(self, keep_tz=False):
                       index=self._shallow_copy(),
                       name=self.name)
 
-    def to_frame(self, index=True, keep_tz=False):
-        """
```

Comment:
is this just a code dupe?

Reply:
If the keep_tz keyword is removed, then yes, this is duplicate code. 

See https://github.com/pandas-dev/pandas/pull/17815#discussion_r143396690 on removing that keyword or not (you merged the PR before my objection was dismissed or adapted :-))

Reply:
hmm do we need to deprecate?

Reply:
nope, because you only merged it a few hours ago (@gfyoung was adding a new method).
(we might consider deprecating it for `Index.to_series`, but that is another issue.

Reply:
I'm not sure I agree with the choice to remove this argument.  It could be confusing to people as to why we support it in one method (`to_series`) but not the other.  Also, your argument as a historical argument was not particularly convincing since you were not 100% sure yourself.

Reply:
OK, maybe I am not 100% of the historical reason, but can you give me one reason why somebody should use this argument?

Reply:
Additionally, the default of `keep_tz=False` is a bad default (no reason it could not preserve the timezone), and so if choosing the keep the keyword but with a different default, it is also not consistent.

Historical argument is correct I think: https://github.com/pandas-dev/pandas/issues/6032

Reply:
@jorisvandenbossche : Full disclosure: I don't use `to_series` and `to_frame` super often, so my changes in that PR were largely for consistency purposes.  After all, that was the point of the original issue: consistency.  Thus, I cannot comment on the utility of `keep_tz` *per se*.

That being said, my points regarding potential user confusion still stand.  We should remove / deprecate the argument uniformly if you are that against it.

Reply:
We are introducing here a *new* method. We should not give a new method suboptimal behaviour that is perfectly avoidable just because of historical reasons (and then I am not even speaking about having the keyword or not, but about it's default).

That said, I am perfectly fine with deprecating the keyword in `to_series` as well, but that will be a backwards incompatible api change (not that I am too worried about it, but technically it is). So that just needs some more general consideration.

Reply:
If you think that `keep_tz` should not be a valid argument, then let's deprecate across the board.  I would be okay with that.

Reply:
To illustrate my 'suboptimal behaviour' of above (this is current master):

```
In [1]: dtidx = pd.date_range("2017-01-01", periods=3, tz='Europe/Brussels')

In [2]: dtidx
Out[2]: 
DatetimeIndex(['2017-01-01 00:00:00+01:00', '2017-01-02 00:00:00+01:00',
               '2017-01-03 00:00:00+01:00'],
              dtype='datetime64[ns, Europe/Brussels]', freq='D')

In [3]: dtidx.to_frame()
Out[3]: 
                                            0
2017-01-01 00:00:00+01:00 2016-12-31 23:00:00
2017-01-02 00:00:00+01:00 2017-01-01 23:00:00
2017-01-03 00:00:00+01:00 2017-01-02 23:00:00

In [5]: dtidx.to_frame().dtypes
Out[5]: 
0    datetime64[ns]
dtype: object
```

so it was a timezone aware index, but it lost its timezone information in conversion to a dataframe.

Reply:
I probably won't have time today, so will already merge this one. 
I might have time later this week, but that will depend on when the RC is cut.

Reply:
Opened an issue for the other one: https://github.com/pandas-dev/pandas/issues/17832